File: Python/1926. Nearest Exit from Entrance in Maze.py
from collections import deque
from typing import List


class Solution:
    def nearestExit(self, maze: List[List[str]], entrance: List[int]) -> int:
        width, length = len(maze), len(maze[0])
        quits = deque()
        quits.append([entrance[0], entrance[1], -1])

        while quits:
            x, y, steps = quits.popleft()

            if not (0 <= x < width and 0 <= y < length):
                if steps > 0:
                    return steps
                continue

            if maze[x][y] == '+':
                continue

            maze[x][y] = '+'
            for _x, _y in ((0, 1), (0, -1), (1, 0), (-1, 0)):
                quits.append([x + _x, y + _y, steps + 1])

        return -1

File: PHP/1926. Nearest Exit from Entrance in Maze.php
<?php

/**
 *  @noinspection PhpIllegalPsrClassPathInspection, PhpMultipleClassDeclarationsInspection
 */
class Solution
{
    private array $entrance;

    private int $width;

    private int $length;

    public function __construct()
    {
        $this->entrance = [];
        $this->width = 0;
        $this->length = 0;
    }

    public function isExit(int $x, int $y): bool
    {
        if ($x === 0 || $y === 0 || $x === $this->width - 1 || $y === $this->length - 1) {
            return $x !== $this->entrance[0] || $y !== $this->entrance[1];
        }

        return false;
    }

    /**
     * @param  string[][]  $maze
     * @param  int[]  $entrance
     */
    public function nearestExit(array $maze, array $entrance): int
    {
        $this->width = count($maze);
        $this->length = count($maze[0]);
        $this->entrance = $entrance;

        $quits = [$entrance];
        $maze[$entrance[0]][$entrance[1]] = 'x';
        $steps = 0;

        $dirs = [
            [-1, 0],
            [1, 0],
            [0, -1],
            [0, 1],
        ];

        while (count($quits)) {
            $qLen = count($quits);

            for ($i = 0; $i < $qLen; $i++) {
                [$x, $y] = array_shift($quits);

                if ($this->isExit($x, $y)) {
                    return $steps;
                }

                for ($j = 0; $j < count($dirs); $j++) {
                    $nx = $x + $dirs[$j][0];
                    $ny = $y + $dirs[$j][1];

                    if ($nx >= 0 && $nx < $this->width && $ny >= 0 && $ny < $this->length && $maze[$nx][$ny] === '.') {
                        $maze[$nx][$ny] = 'x';
                        $quits[] = [$nx, $ny];
                    }
                }
            }
            $steps++;
        }

        return -1;
    }
}
